NDEV-3549 do not add extra accounts for precompilted constracts calls

artem-yazkov opened the pull request.

artem-yazkov committed:
NDEV-3549 do not add extra accounts for precompilted constracts

artem-yazkov committed:
NDEV-3549 zero precompiled contracts on backend.code_hash()

artem-yazkov committed:
NDEV-3549 do not add extra accounts for precompile addresses, precompile extension addresses

NDEV-3549 some debug logs added

artem-yazkov committed:
NDEV-3549 do not skip precompile contract addresses

neonlabstech commented:
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/13241378900/artifacts/2564580450)

neonlabstech commented:
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/13241919384/artifacts/2564733495)

artem-yazkov committed:
NDEV-3549 return empty code for precompile addresses

NDEV-3549 return '0xFE' code for precompile extension addresses

NDEV-3549 revert d71b6f7c5525ed7314fa3d36f86ca6434d1cc205
"zero precompiled contracts on backend.code_hash()"

artem-yazkov committed:
NDEV-3549 remove debug logs

NDEV-3549 don't touch contracts on ExecutorState::transfer()

neonlabstech commented:
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/13309328420/artifacts/2586166343)

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |   48488 |      0     |      0.228 |    0.00109241  |         0.001 |
| getBlocks                         |   29910 |      0     |      0.237 |    0.00157018  |         0.001 |
| getBlock                          |   15324 |      0     |      0.059 |    0.00187843  |         0.001 |
| getTransaction                    |   10708 |      0     |      0.284 |    0.0128341   |         0.004 |
| getMultipleAccounts               |    8003 |      0     |      0.239 |    0.00218443  |         0.001 |
| sendTransaction                   |    5235 |      0.001 |      0.238 |    0.0130382   |         0.006 |
| getLatestBlockhash                |    4793 |      0     |      0.398 |    0.00529668  |         0.002 |
| getBalance                        |    1870 |      0     |      0.075 |    0.0013492   |         0.001 |
| getSignatureStatuses              |     420 |      0     |      0.058 |    0.00100952  |         0.001 |
| simulateTransaction               |     273 |      0     |      0.007 |    0.00217582  |         0.002 |
| isBlockhashValid                  |     192 |      0     |      0.009 |    0.000640625 |         0     |
| getVersion                        |     168 |      0     |      0.047 |    0.00127976  |         0.001 |
| getSignaturesForAddress           |      81 |      0     |      0.026 |    0.00234568  |         0.002 |
| requestAirdrop                    |      32 |      0.001 |      0.002 |    0.00109375  |         0.001 |
| getSlot                           |      25 |      0     |      0.006 |    0.00136     |         0.001 |
| getBlockHeight                    |      15 |      0     |      0.015 |    0.002       |         0.001 |
| getHealth                         |       4 |      0.001 |      0.29  |    0.07325     |         0.001 |
| getTokenAccountBalance            |       3 |      0     |      0.001 |    0.000333333 |         0     |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |         0.001 |
| getProgramAccounts                |       1 |      0.045 |      0.045 |    0.045       |         0.045 |
| getClusterNodes                   |       1 |      0.001 |      0.001 |    0.001       |         0.001 |

</details>
